pennylane/measurements/shots.py

```diff
@@ -192,6 +192,11 @@ def __hash__(self):
         """Hash for a given Shot instance."""
         return hash(self.shot_vector)
 
+    def __iter__(self):
+        for shot_copy in self.shot_vector:
+            for _ in range(shot_copy.copies):
+                yield shot_copy.shots
+
```

Comment:
We should have some unit tests for this in `tests/measurements/test_shots.py`.

Reply:
Added unit tests.